applications/ConvectionDiffusionApplication/custom_utilities/bfecc_convection.h

```diff
@@ -68,9 +68,15 @@ template<std::size_t TDim> class BFECCConvection
         std::vector< Vector > Ns( rModelPart.Nodes().size());
         std::vector< bool > found( rModelPart.Nodes().size());
 
-        // Allocate non-historical variables
-        for (auto &r_node : rModelPart.Nodes()) {
-            r_node.SetValue(rVar, 0.0);
+        // Allocate non-historical variables and update old velocity as per dt_factor
+        #pragma omp parallel for
+        for (int i = 0; i < nparticles; i++) {
+            auto iparticle = rModelPart.NodesBegin() + i;
+            iparticle->SetValue(rVar, 0.0);
+            const auto old_velocity = iparticle->FastGetSolutionStepValue(conv_var, 1);
+            const auto current_velocity = iparticle->FastGetSolutionStepValue(conv_var);
```

Comment:
```suggestion
            const auto& r_current_velocity = iparticle->FastGetSolutionStepValue(conv_var);
```
You're copying the value.